lib/online_order.rb

```diff
@@ -0,0 +1,117 @@
+#
+# self.all - returns a collection of OnlineOrder instances, representing all of the OnlineOrders described in the CSV. See below for the CSV file specifications
+# Question Ask yourself, what is different about this all method versus the Order.all method? What is the same?
+
+#self.find(id) - returns an instance of OnlineOrder where the value of the id field in the CSV matches the passed parameter. -Question Ask yourself, what is different about this find method versus the Order.find method?
+
+# self.find_by_customer(customer_id) - returns a list of OnlineOrder instances where the value of the customer's ID matches the passed parameter.
+
+require_relative 'order'
+require 'csv'
+require 'pry'
+
+module Grocery
+  class OnlineOrder < Order
+    attr_reader :customer_id
+    attr_accessor :status
+    def initialize(id, products, customer_id, status = :pending)
+      super(id, products)
+      @customer_id = customer_id
+      @status = status.to_sym
+      @products = products
+    end
+
+    def self.all
+      @online_orders = []
+      CSV.open("support/online_orders.csv", 'r').each do |line|
+
+        id = line[0].to_i
+        customer_id = line[2].to_i
+        status = line[3].to_sym
+        products_hash = {}
+        products_arr = line[1].split';'
+
+        products_arr.each do |item_and_price|
+          product_price = item_and_price.split':'
+          products_hash[product_price[0]] = product_price[1].to_f
+        end
+        @online_orders << self.new(id, products_hash, customer_id, status)
+      end
+      return @online_orders
+    end
+
+    def self.find(val)
+      all_online_orders = OnlineOrder.all
+      found_order = nil
+
+      all_online_orders.each do |order|
+        if order.id == val
+          found_order = order
+        end
+      end
+      if found_order == nil
+        raise ArgumentError.new("Invalid order id #{val}")
+      else
+        return found_order
+      end
+    end
+
+    def self.find_by_customer(val)
+      all_online_orders = Grocery::OnlineOrder.all
+
+      customer_order = nil
+      customer_orders = []
+
+      all_online_orders.each do |order|
+        if order.customer_id == val
+          customer_order = order
+          customer_orders << customer_order
+        end
+      end
+      if customer_order == nil
+        raise ArgumentError.new("Invalid customer id #{val}")
+      else
+        return customer_orders
+      end
+    end
+
+    def total
+      if @products ==  nil || @products == 0
+        return 0
+      else
+        super + 10 #adds $10 shipping fee
+      end
+    end
+
+    def add_product(product_name, product_price)
+      unless @status == :pending || @status == :paid
+        raise ArgumentError.new("Due to order status of #{@status}, additional products can no longer be added to the order.")
+      end
+      if @products.include?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
```

Comment:
Would it be possible to reuse the implementation of `add_product` in the `Order` class via `super` here instead of typing it all out?